### PR TITLE
Plugin flow builder: tracking

### PR DESCRIPTION
**Pull request by Iru89**
## Description
Add the possibility to pass a function from the bot configuration to do the tracking.
- track the name of each first content, not the "followups"
- track the state of the queue as it passes through a conditional queue. At this point it would be interesting for the queue status component to store the name of the project. 
- track HANDOFF_SUCESSFULL when doing a handoff

## Testing

The pull request...

- has unit tests
- has integration tests
- doesn't need tests because... **[provide a description]**

### CODE REVIEW

File: packages/botonic-plugin-flow-builder/src/action.tsx [hunk 1]
```diff
@@ -46,26 +46,28 @@ export class FlowBuilderAction extends React.Component<FlowBuilderActionProps> {
       locale
     )
 
+    if (flowBuilderPlugin.trackEvent) {
+      // TODO: track all targets nodes?
```
Comment (vanbasten17) [inline packages/botonic-plugin-flow-builder/src/action.tsx:50]: @asastre should we track just the "parent" or all the follow-up/target contents?

File: packages/botonic-plugin-flow-builder/src/functions/conditional-queue-status.ts [hunk 2]
```diff
@@ -26,5 +34,18 @@ export async function conditionalQueueStatus({
     }
   )
   const isAvailable = response.data.available
+
+  console.log('conditionalQueueStatus', { data: response.data })
```
Comment (vanbasten17) [inline packages/botonic-plugin-flow-builder/src/functions/conditional-queue-status.ts:38]: Remove console log

File: packages/botonic-plugin-flow-builder/src/functions/conditional-queue-status.ts [hunk 3]
```diff
@@ -26,5 +34,18 @@ export async function conditionalQueueStatus({
...
+  const flowBuilderPlugin = getFlowBuilderPlugin(request.plugins)
+  if (flowBuilderPlugin.trackEvent) {
+    const eventName = 'QUEUE_STATUS'
+    const args = {
+      queue_id,
+      queue_name,
+      status: isAvailable ? 'open' : 'closed',
+    }
+    await flowBuilderPlugin.trackEvent(request, eventName, args)
```
Comment (vanbasten17) [inline packages/botonic-plugin-flow-builder/src/functions/conditional-queue-status.ts:47]: For next iterations, I would use the decorator pattern to automatically call this, but atm is quite overengineering. Let's if we need to extend this logic in the future

File: packages/botonic-plugin-flow-builder/src/handoff.ts [hunk 4]
```diff
@@ -36,15 +42,19 @@ export async function doHandoff(
   // }
 
   await handOffBuilder.handOff()
+
+  if (flowBuilderPlugin.trackEvent) {
+    await flowBuilderPlugin.trackEvent(request, 'HANDOFF_SUCCESSFULL', {
```
Comment (vanbasten17) [inline packages/botonic-plugin-flow-builder/src/handoff.ts:47]: What about passing also the list of enums to track event?

**Review by vanbasten17 — APPROVED** [review]
@Iru89 you rock!! Thanks @Iru89 🤗

File: packages/botonic-plugin-flow-builder/src/handoff.ts [hunk 5]
```diff
@@ -36,15 +42,19 @@ export async function doHandoff(
...
+      queueName: queueFound?.name,
```
Comment (AinaVendrell) [inline packages/botonic-plugin-flow-builder/src/handoff.ts:48]: i think this event should be more similar to the `QUEUE_STATUS` event.
One only tracks the `queueName` and the other the `queue_id` and `queue_name`. Also one is in camel case and the other in snake case

File: packages/botonic-plugin-flow-builder/src/handoff.ts (hunk 4 above)
Comment (Iru89) [reply packages/botonic-plugin-flow-builder/src/handoff.ts:47]: I think it is better to have default events that are always the same. Maybe we could pass a list optionally and if it is not passed we could use the default events.

Comment (vanbasten17) [reply packages/botonic-plugin-flow-builder/src/handoff.ts:47]: cool! 👌

**Review by AinaVendrell — APPROVED** [review]

File: packages/botonic-plugin-flow-builder/src/action.tsx (hunk 1 above)
Comment (vanbasten17) [reply packages/botonic-plugin-flow-builder/src/action.tsx:50]: cool! let's iterate it for next sprints

**Pull request merged**